aie_runtime_lib/AIE2/sqrt.h

```diff
@@ -0,0 +1,64 @@
+//===-  sqrt.h - get square root values for bfloat16 data type
+//-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2023 Advanced Micro Devices, Inc.
+//
+//
+//===----------------------------------------------------------------------===//
+// This is the implementation of compute square root values for bfloat16 type
+//===----------------------------------------------------------------------===//
+
+#ifndef __SQRT_H__
+#define __SQRT_H__
+
+#include "aie_api/aie.hpp"
+#include <aie_api/aie_adf.hpp>
+#include <aie_api/utils.hpp>
+
+v32bfloat16 getSqrtBf16(v32bfloat16 in) {
+  aie::vector<bfloat16, 32> x = in;
+  aie::accum<accfloat, 32> x2 =
+      aie::mul(x, bfloat16(0.5f)); // x2 = number * 0.5F;
+  const aie::vector<int16, 32> magic = aie::broadcast<int16, 32>(0x5f37);
+  aie::vector<bfloat16, 32> y =
+      aie::sub(magic, aie::downshift(x.cast_to<int16>(), 1))
+          .cast_to<bfloat16>();
+
+  x2 = aie::mul(x2.to_vector<bfloat16>(), y); // x2 * y
```

Comment:
I find that the comments at the end of each line are not very readable. May this is caused by clang-tidy.
It would be better to have a comment block immediately after the open { of the function body that describes the algorithm.